Index: FreeRDP-Remmina-356c033/remmina/desktop/CMakeLists.txt
===================================================================
--- FreeRDP-Remmina-356c033.orig/remmina/desktop/CMakeLists.txt
+++ FreeRDP-Remmina-356c033/remmina/desktop/CMakeLists.txt
@@ -38,3 +38,4 @@ install(FILES ${ICON32_DATA} DESTINATION
 install(FILES ${ICON48_DATA} DESTINATION ${ICON48_DIR})
 install(FILES ${ICONSVG_DATA} DESTINATION ${ICONSVG_DIR})
 
+install(FILES remmina.desktop DESTINATION "${REMMINA_DATADIR}/applications")
Index: FreeRDP-Remmina-356c033/remmina/desktop/remmina.desktop
===================================================================
--- /dev/null
+++ FreeRDP-Remmina-356c033/remmina/desktop/remmina.desktop
@@ -0,0 +1,13 @@
+[Desktop Entry]
+Version=1.0
+Name=Remmina
+GenericName=Remote Desktop Client
+X-GNOME-FullName=Remmina Remote Desktop Client
+Comment=Connect to remote desktops
+TryExec=remmina
+Exec=remmina
+Icon=remmina
+Terminal=false
+Type=Application
+Categories=GTK;GNOME;X-GNOME-NetworkSettings;Network;
+Actions=Profile;Tray;
